pkg/apis/sources/v1alpha1/ibmmq_types.go

```diff
@@ -74,6 +74,7 @@ type Credentials struct {
 type TLSSpec struct {
 	Cipher             string   `json:"cipher"`
 	ClientAuthRequired bool     `json:"clientAuthRequired"`
+	CertLabel          string   `json:"certLabel"`
```

Comment:
I didn't make it a pointer type to spare additional `nil` checks in the adapter since empty strings are totally acceptable values in the underlying functions. If there are some k8s best practices on that matter that I missed, I'd gladly accept them because I often change my decisions depending on the particular case. E.g. in other cases, I'd make it a pointer in combination with `omitempty` tag to make pod's env smaller.

Reply:
Or I guess in this case nil pointer will result empty string after env decoding anyway

Reply:
The pointer for optional is a Kubernetes API convention which is described here: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required

We typically try to follow these conventions, being a Kubernetes-based product.

In this case it doesn't technically make a difference, but it could if we introduce an OpenAPI validation on that field. For example, `minLength: 1` would fail if the field is `string` and empty, but not if the field is `*string` and empty, because Kubernetes will eventually serialize to `""` and that will be rejected. (`omitempty` might be enough to fix that in the case of a `string` field though)

When passed to `envconfig`, it will be an empty string in any case.

Reply:
@antoineco thanks for the link!
The fact that the `string` type in this particular case didn't make a difference but it did let me skip nil check was my driver for this decision. However, if you like b066090 more, let it be the pointer.